src/hotspot/share/gc/shared/stringdedup/stringDedupTable.cpp

```diff
@@ -395,31 +395,33 @@ bool StringDedupTable::is_rehashing() {
 StringDedupTable* StringDedupTable::prepare_resize() {
   size_t size = _table->_size;
 
-  // Check if the hashtable needs to be resized
+  // Decide whether to resize, and compute desired new size if so.
   if (_table->_entries > _table->_grow_threshold) {
-    // Grow table, double the size
-    size *= 2;
-    if (size > _max_size) {
-      // Too big, don't resize
-      return NULL;
+    // Compute new size.
+    size_t needed = _table->_entries / _grow_load_factor;
+    if (needed < _max_size) {
+      size = round_up_power_of_2(needed);
+    } else {
+      size = _max_size;
```

Comment:
It's not obvious to me which one is larger, btw `round_up_power_of_2(needed)` and `_max_size`.

Reply:
This table implementation uses power of 2 sizes, including _min/max_size.
That's assumed throughout.  Since needed < _max_size, rounding needed up to
a power of 2 can't exceed _max_size.

Reply:
I see; thanks for the explanation.